fabric_generator/file_parser.py

```diff
@@ -0,0 +1,641 @@
+
+oppositeDic = {"NORTH": "SOUTH", "SOUTH": "NORTH",
+               "EAST": "WEST", "WEST": "EAST"}
+
+
+def parseFabricCSV(fileName: str) -> Fabric:
+    """
+    Parses a csv file and returns a Fabric object.
+    """
+
+    with open(fileName, 'r') as f:
+        file = f.read()
+        file = re.sub(r"#.*", "", file)
+
+    # read in the csv file and part them
+    if fabricDescription := re.search(
+            r"FabricBegin(.*?)FabricEnd", file, re.MULTILINE | re.DOTALL):
+        fabricDescription = fabricDescription.group(1)
+    else:
+        raise ValueError(
+            'ERROR: cannot find FabricBegin and FabricEnd in csv file')
+
+    if parameters := re.search(
+            r"ParametersBegin(.*?)ParametersEnd", file, re.MULTILINE | re.DOTALL):
+        parameters = parameters.group(1)
+    else:
+        raise ValueError(
+            'ERROR: cannot find ParametersBegin and ParametersEnd in csv file')
+
+    tilesData = re.findall(r"TILE(.*?)EndTILE", file,
+                           re.MULTILINE | re.DOTALL)
+
+    superTile = re.findall(r"SuperTILE(.*?)EndSuperTILE",
+                           file, re.MULTILINE | re.DOTALL)
+
+    # parse the tile description
+    fabricDescription = fabricDescription.split("\n")
+    parameters = parameters.split("\n")
+    tileTypes = []
+    tileDefs = []
+    commonWirePair: List[Tuple[str, str]] = []
+    for t in tilesData:
+        t = t.split("\n")
+        tileName = t[0].split(",")[1]
+        tileTypes.append(tileName)
+        ports: List[Port] = []
+        bels: List[Bel] = []
+        matrixDir = ""
+        withUserCLK = False
+        for item in t:
+            temp: List[str] = item.split(",")
+            if not temp or temp[0] == "":
+                continue
+            if temp[0] in ["NORTH", "SOUTH", "EAST", "WEST"]:
+                ports.append(Port(Direction[temp[0]], temp[1], int(
+                    temp[2]), int(temp[3]), temp[4], int(temp[5]), temp[1], IO.OUTPUT, Side[temp[0]]))
+
+                ports.append(Port(Direction[temp[0]], temp[1], int(
+                    temp[2]), int(temp[3]), temp[4], int(temp[5]), temp[4], IO.INPUT, Side[oppositeDic[temp[0]].upper()]))
+                # wireCount = (abs(int(temp[2])) +
+                #              abs(int(temp[3])))*int(temp[5])
+                # for i in range(wireCount):
+                commonWirePair.append(
+                        (f"{temp[1]}", f"{temp[4]}"))
+
+            elif temp[0] == "JUMP":
+                ports.append(Port(Direction.JUMP, temp[1], int(
+                    temp[2]), int(temp[3]), temp[4], int(temp[5]), temp[1], IO.OUTPUT, Side.ANY))
+                ports.append(Port(Direction.JUMP, temp[1], int(
+                    temp[2]), int(temp[3]), temp[4], int(temp[5]), temp[1], IO.INOUT, Side.ANY))
+            elif temp[0] == "BEL":
+                if temp[1].endswith(".vhdl"):
+                    result = parseFileVHDL(temp[1], temp[2])
```

Comment:
I think this needs to be computing a path relative to `fabric.csv` here - this is failing with a file not found when testing out the flow personally